ui/src/app/workflows/components/resubmit-workflow-panel.tsx

```diff
@@ -12,103 +12,93 @@ interface Props {
     isArchived: boolean;
 }
 
-interface State {
-    overrideParameters: boolean;
-    workflowParameters: Parameter[];
-    memoized: boolean;
-    error?: Error;
-    isSubmitting: boolean;
-}
+export function ResubmitWorkflowPanel(props: Props) {
+    const [overrideParameters, setOverrideParameters] = useState(false);
+    const [workflowParameters, setWorkflowParameters] = useState<Parameter[]>(JSON.parse(JSON.stringify(props.workflow.spec.arguments.parameters || [])));
+    const [memoized, setMemoized] = useState(false);
+    const [error, setError] = useState<Error>();
+    const [isSubmitting, setIsSubmitting] = useState(false);
 
-export class ResubmitWorkflowPanel extends React.Component<Props, State> {
-    constructor(props: any) {
-        super(props);
-        const state: State = {
-            workflowParameters: JSON.parse(JSON.stringify(this.props.workflow.spec.arguments.parameters || [])),
-            memoized: false,
-            isSubmitting: false,
-            overrideParameters: false
+    const submit = () => {
+        setIsSubmitting(true);
+        const parameters: ResubmitOpts['parameters'] = overrideParameters
+            ? [...workflowParameters.filter(p => Utils.getValueFromParameter(p) !== undefined).map(p => p.name + '=' + Utils.getValueFromParameter(p))]
+            : [];
+        const opts: ResubmitOpts = {
+            parameters,
+            memoized
         };
-        this.state = state;
-    }
 
-    public render() {
-        return (
-            <>
-                <h4>Resubmit Workflow</h4>
-                <h5>
-                    {this.props.workflow.metadata.namespace}/{this.props.workflow.metadata.name}
-                </h5>
-                {this.state.error && <ErrorNotice error={this.state.error} />}
-                <div className='white-box'>
-                    <div key='override-parameters' style={{marginBottom: 25}}>
-                        <label>Override Parameters</label>
-                        <div className='columns small-9'>
-                            <Checkbox checked={this.state.overrideParameters} onChange={overrideParameters => this.setState({overrideParameters})} />
-                        </div>
-                    </div>
+        const handleSuccess = (submitted: Workflow) => {
+            document.location.href = uiUrl(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`);
```

Comment:
this is the same as it was before, but shouldn't this use react-router calls instead?

Reply:
Ah I think the current pattern is to use the History API with `history.push`. We are on an older version of react-router though (I'm working on upgrading it -- have to finish removing `BasePage` first), so I don't think the `useHistory` hook is available. Other components use `history` from props currently

Reply:
thanks! I was just researching how to write in current react router version.
I'll try!

Reply:
I tryed passing `history` in props and replaced 'document.location.href' with 'history.push', but it didn't work as I expected :(

```props.history.push(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`);```

Reply:
If the scope of the changes is likely to be larger than initially expected, it might be good to reconsider the approach after your react-router version up I think.

Reply:
> `props.history.push(`workflows/${submitted.metadata.namespace}/${submitted.metadata.name}`);`

you do still need `uiUrl`, as in `props.history.push(uiUrl("..."))`. though if `BASE_HREF` isn't set (and it's not when running locally), all it does is [add a `/`](https://github.com/argoproj/argo-workflows/blob/55bb51885d2a6690727f97dce25fffef1afb34f2/ui/src/app/shared/base.ts#L3) to the beginning.

I'm not sure if that fixes this?

> but it didn't work as I expected :(

how did it end up working? just so we know for future reference next time somebody tries tackling this.

> If the scope of the changes is likely to be larger than initially expected, it might be good to reconsider the approach after your react-router version up I think.

Yea I'm good approving as is, though if you can figure it out, would be a nice addition too 🙂

Reply:
Thanks a lot!!
by `props.history.push(uiUrl("..."))`,  it works for resubmit and retry component!
but in the resubmit case, the icon of the node in the workflow graph at the destination of the location does not change.
before the modification, the icon of the node in the graph panel had changed to progress status.

<img width="80" alt="Screen Shot 2023-09-14 at 22 04 05" src="https://github.com/argoproj/argo-workflows/assets/51753414/e53045a7-0af8-4309-ba8c-5535927384a2">

I confirmed that the workflow pod outputs logs, so I think the workflow itself was working.
do I need change some state of the workflow graph node?

and in the retry case, this degradation doesnt occur. 🤒

Reply:
> Does it eventually update itself to in progress?

no, the icon does not change at all from this image
<img width="109" alt="Screen Shot 2023-09-14 at 22 33 50" src="https://github.com/argoproj/argo-workflows/assets/51753414/ed8b5df4-1e2a-418b-9459-d7058c5775b7">

before the modification, the icon had changed yellow -> blue -> green

Reply:
Huh. Since this is on the same page (just different Workflow), I wonder if maybe the page is not quite unloading correctly and still has the old state. The effects seem to have [proper disposers](https://github.com/argoproj/argo-workflows/blob/7aedf973356c8b57510a554b6b759f2684f88839/ui/src/app/workflows/components/workflow-details/workflow-details.tsx#L386) and [dependencies](https://github.com/argoproj/argo-workflows/blob/7aedf973356c8b57510a554b6b759f2684f88839/ui/src/app/workflows/components/workflow-details/workflow-details.tsx#L427), so I don't see anything strange off the top of my head.

At this point, especially since it potentially involves a whole other page's logic, I think we should punt on the navigation improvements for now. Appreciate your efforts and investigation here! Hopefully that will help the next person to tackle this too (which could be yourself 😉 ).

Reply:
In the retry component, the state appears to be changing as expected, so the difference in implementation between retry and resubmit may be a clue, though I haven't been able to found out.

Reply:
If I figure it out, I'll make a PR🔥

Reply:
> so the difference in implementation between retry and resubmit may be a clue

So a retry does not create a new Workflow, it just reruns failed steps. A resubmit creates a wholly new Workflow, effectively a clone of the old one.

So for a retry, the URL does not change. For a resubmit it does.
For the retry, the existing `ListWatch` might be able to pick up the change automatically. For a resubmit, an entirely new request is needed for the new Workflow.

Maybe those details may help you debug?
I already knew that though (I wrote a bit of the documentation on retry vs resubmit in #11625) and couldn't figure it out, though I didn't stare too hard